catalyst/dl/experiment/gan.py

```diff
@@ -0,0 +1,158 @@
+                will be wrapped for "generator_train" phase
+                in wrap_callbacks method
+        """
+        super().__init__(
+            model,
+            loaders,
+            callbacks,
+            logdir,
+            stage,
+            criterion,
+            optimizer,
+            scheduler,
+            num_epochs,
+            valid_loader,
+            main_metric,
+            minimize_metric,
+            verbose,
+            state_kwargs,
+            checkpoint_data,
+            distributed_params,
+            monitoring_params,
+            initial_seed,
+        )
+        self.wrap_callbacks(phase2callbacks)
+
+    def wrap_callbacks(self, phase2callbacks) -> None:
+        """Phase wrapping procedure for callbacks"""
+        discriminator_phase_name = self._additional_state_kwargs[
+            "discriminator_train_phase"
+        ]
+        discriminator_phase_num = self._additional_state_kwargs[
+            "discriminator_train_num"
+        ]
+        generator_phase_name = self._additional_state_kwargs[
+            "generator_train_phase"
+        ]
+        generator_phase_num = self._additional_state_kwargs[
+            "generator_train_num"
+        ]
+        self._callbacks["phase_manager"] = PhaseManagerCallback(
+            train_phases=OrderedDict(
+                [
+                    (discriminator_phase_name, discriminator_phase_num),
+                    (generator_phase_name, generator_phase_num),
+                ]
+            ),
+            valid_mode="all",
+        )
+        for phase_name, callback_name_list in phase2callbacks.items():
+            # TODO: Check for phase in state_params?
+            for callback_name in callback_name_list:
+                callback = self._callbacks.pop(callback_name)
+                self._callbacks[callback_name] = PhaseBatchWrapperCallback(
+                    base_callback=callback, active_phases=[phase_name],
+                )
+
+    def get_callbacks(self, stage: str) -> "OrderedDict[str, Callback]":
+        callbacks = super().get_callbacks(stage="")
+        default_callbacks = []
+        if self._verbose:
+            default_callbacks.append(("verbose", VerboseLogger))
+        if not stage.startswith("infer"):
+            default_callbacks.append(("saver", CheckpointCallback))
+            default_callbacks.append(("console", ConsoleLogger))
+        default_callbacks.append(("exception", RaiseExceptionCallback))
+        # Check for absent callbacks and add them
+        for callback_name, callback_fn in default_callbacks:
+            is_already_present = any(
+                isinstance(x, callback_fn) for x in callbacks.values()
+            )
+            if not is_already_present:
+                callbacks[callback_name] = callback_fn()
+        return callbacks
+
+    def get_transforms(self, stage: str = None, dataset: str = None):
+        raise NotImplementedError(
```

Comment:
maybe just `pass`? 🤔 
what do you think @Scitator @bagxi @lightforever ?

Reply:
The error string description is probably misleading, @Arquestro what did you mean here?
"No transforms are used in GAN experiment" looks... clearly wrong as you may and should use them.

I suppose you can just do not implement `get_transforms` to force implementation in descendants or modify error message to make it clear why this method is not implemented

Reply:
@TezRomacH I think raising error is more safe here than `pass`

Reply:
Every pull request gives state for some functionality.
There are no subclasses of `GanExperiment` or routines that will use it's transforms getter, so raising `NotImplementedError` is right, because it's the part of GanExperiment state for following release.
What I agree about, is misguiding message of the exception, will change it.

Reply:
@Arquestro why do we need to raise an error here?

Reply:
@Arquestro could you please explain why do you need `get_transforms` now? seems you only re-raising exception with another message
maybe it is better to remove `get_transforms`, like in `SupervisedExperiment`?

Reply:
@bagxi only to reraise it with another message, the only reason. And to remove PyCharms annoying warning about implementation of all abstract methods.

Reply:
Removed this method.